File: algorithmAndDataStructure/golang/climbingStairs/climbingStairs.go
package climbingStairs

func climbStairs(n int) int {
	lastOneStepWays, lastTwoStepWays := 1, 0
	for i := 1; i < n; i++ {
		lastOneStepWays, lastTwoStepWays = lastOneStepWays+lastTwoStepWays, lastOneStepWays
	}
	return lastOneStepWays + lastTwoStepWays
}

File: algorithmAndDataStructure/cpp/climbingStairs/climbingStairs.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        int lastOneStepWays = 1, lastTwoStepWays = 0;
        for (int i = 1; i < n; i++) {
            int ways = lastTwoStepWays + lastOneStepWays;
            lastTwoStepWays = lastOneStepWays;
            lastOneStepWays = ways;
        }
        return lastOneStepWays + lastTwoStepWays;
    }
};